pkg/controllers/generic/from_host_syncer.go

```diff
@@ -0,0 +1,306 @@
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+type FromHostSyncer interface {
+	SyncToHost(vOjb, pObj client.Object)
+	GetProPatches(ctx *synccontext.SyncContext) []config.TranslatePatch
+	GetMappings(ctx *synccontext.SyncContext) map[string]string
+}
+
+func NewFromHost(ctx *synccontext.RegisterContext, fromHost FromHostSyncer, translator syncertypes.FromConfigTranslator, skipFuncs ...translator.ShouldSkipHostObjectFunc) (syncertypes.Object, error) {
+	s := &genericFromHostSyncer{
+		FromHostSyncer:       fromHost,
+		FromConfigTranslator: translator,
+		skipFuncs:            skipFuncs,
+	}
+	virtualToExclude := s.checkExperimentalDeployConfig(ctx)
+	s.virtualObjectsToExclude = virtualToExclude
+	return s, nil
+}
+
+type genericFromHostSyncer struct {
+	syncertypes.FromConfigTranslator
+	FromHostSyncer
+	skipFuncs               []translator.ShouldSkipHostObjectFunc
+	virtualObjectsToExclude map[string]bool
+}
+
+func (s *genericFromHostSyncer) Options() *syncertypes.Options {
+	return &syncertypes.Options{
+		UsesCustomPhysicalCache: true,
+	}
+}
+
+func (s *genericFromHostSyncer) SyncToHost(ctx *synccontext.SyncContext, event *synccontext.SyncToHostEvent[client.Object]) (ctrl.Result, error) {
+	klog.FromContext(ctx).V(1).Info("SyncToHost called")
+	return ctrl.Result{}, ctx.VirtualClient.Delete(ctx, event.Virtual)
+}
+
+func (s *genericFromHostSyncer) Sync(ctx *synccontext.SyncContext, event *synccontext.SyncEvent[client.Object]) (_ ctrl.Result, retErr error) {
+	klog.FromContext(ctx).V(1).Info("Sync called")
+
+	patchHelper, err := patcher.NewSyncerPatcher(ctx, event.Host, event.Virtual, patcher.TranslatePatches(s.GetProPatches(ctx), false))
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("new syncer patcher: %w", err)
+	}
+
+	defer func() {
+		patchHelper.SkipHostPatch()
+		if err := patchHelper.Patch(ctx, event.Host, event.Virtual); err != nil {
+			retErr = utilerrors.NewAggregate([]error{retErr, err})
+		}
+		if retErr != nil {
+			s.EventRecorder().Eventf(event.Virtual, "Warning", "SyncError", "Error syncing: %v", retErr)
+		}
+	}()
+
+	s.FromHostSyncer.SyncToHost(event.Virtual, event.Host)
+
+	return ctrl.Result{}, nil
+}
+
+func (s *genericFromHostSyncer) SyncToVirtual(ctx *synccontext.SyncContext, event *synccontext.SyncToVirtualEvent[client.Object]) (ctrl.Result, error) {
+	klog.FromContext(ctx).V(1).Info("SyncToVirtual called")
+	if event.VirtualOld != nil || event.Host.GetDeletionTimestamp() != nil {
+		return patcher.DeleteHostObject(ctx, event.Host, event.VirtualOld, "virtual object was deleted")
+	}
+
+	vObj := translate.VirtualMetadata(event.Host, s.HostToVirtual(ctx, types.NamespacedName{Name: event.Host.GetName(), Namespace: event.Host.GetNamespace()}, event.Host))
+
+	err := pro.ApplyPatchesVirtualObject(ctx, nil, vObj, event.Host, s.GetProPatches(ctx), false)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// make sure namespace exists
```

Comment:
nit: I think the namespace check can be moved before the `vObj` variable and `ApplyPatchesVirtualObject` call.

Reply:
:+1: moved